src/main/java/database/DatabaseConnector.java

```diff
@@ -14,149 +14,153 @@
+     * or null if an exception occurred
+     */
+    public Question[] getAllQuestions() {
+        try {
+            Statement stmt = con.createStatement();
+            ResultSet rs = stmt.executeQuery(
+                    "SELECT * FROM " + DEFAULT_TABLE_NAME);
+
+            List<Question> questions = new ArrayList<>();
+            while (rs.next()) {
+                List<String> answers = new ArrayList<>();
+                for (int i = 0; i < 3; i++) {
+                    answers.add(rs.getString(COLUMN_ANSWER[i]));
+                }
+                questions.add(new Question(
+                        rs.getString(COLUMN_QUESTION),
+                        answers));
+
+            }
+            rs.close();
+            stmt.close();
+
+            return questions.toArray(new Question[questions.size()]);
+        } catch (SQLException e) {
+            onSQLException(e);
+            return null;
+        }
+    }
+
+    /**
+     * Removes a question from the database
+     *
+     * @param q The question to be removed
+     */
+    public void removeQuestion(Question q) {
+        try {
+            Statement stmt = con.createStatement();
+            stmt.executeUpdate(
+                    "DELETE FROM " + DEFAULT_TABLE_NAME
+                    + "WHERE "
+                    + "(" + COLUMN_QUESTION + "='" + q.getQuestion() + "');");
+            stmt.close();
+        } catch (SQLException e) {
+            onSQLException(e);
+        }
+    }
+
+    /**
+     * Inserts a new question into the database
+     *
+     * @param q The new question
+     */
+    public void addQuestion(Question q) {
+        try {
+            Statement stmt = con.createStatement();
+
+            //The correct answer is set as the first element in the array
+            Answer[] answer = q.getAnswers();
+            int correct = -1;
+            for (int i = 0; i < answer.length; i++) {
+                if (q.isCorrectAnswer(answer[i].getId())) {
+                    correct = i;
+                    break;
+                }
+            }
+            if (correct != 0) {
+                Answer correctanswer = answer[correct];
+                answer[correct] = answer[0];
+                answer[0] = correctanswer;
+            }
+
+            stmt.executeUpdate(
+                    "INSERT INTO " + DEFAULT_TABLE_NAME
+                    + "(" + COLUMN_QUESTION + ", " + COLUMN_ANSWER[0] + ", " + COLUMN_ANSWER[1] + ", " + COLUMN_ANSWER[2] + ")"
+                    + "VALUES "
+                    + "('" + q.getQuestion() + "', '" + answer[0] + "', '" + answer[1] + "', '" + answer[2] + "');");
+            stmt.close();
+        } catch (SQLException e) {
+            onSQLException(e);
+        }
+    }
+
+    @Override
+    public void close() throws Exception {
+        if (!con.isClosed()) {
+            con.close();
+        }
+    }
+
+    private void onSQLException(SQLException e) {
+        e.printStackTrace();//TODO uncool
```

Comment:
Yep, as you noted, that's still todo. Well, I'm not exactly sure if the error handling works out like this.. Gosh, Java error handling is just super broken IMO.